src/compile/format.ts

```diff
@@ -179,22 +178,16 @@ export function guideFormat(
 ) {
   if (isCustomFormatType(formatType)) {
     return undefined; // handled in encode block
-  } else if (
-    format === undefined &&
-    formatType === undefined &&
-    config.customFormatTypes &&
-    channelDefType(fieldOrDatumDef) === 'quantitative'
-  ) {
+  } else if (format === undefined && formatType === undefined && config.customFormatTypes) {
     if (
-      config.normalizedNumberFormatType &&
-      isPositionFieldOrDatumDef(fieldOrDatumDef) &&
-      fieldOrDatumDef.stack === 'normalize'
+      channelDefType(fieldOrDatumDef) === 'quantitative' &&
+      ((config.normalizedNumberFormatType &&
+        isPositionFieldOrDatumDef(fieldOrDatumDef) &&
+        fieldOrDatumDef.stack === 'normalize') || // case: normalized number format
+        config.numberFormatType) // case: regular number format
     ) {
       return undefined; // handled in encode block
     }
-    if (config.numberFormatType) {
```

Comment:
Why is this block removed?

Reply:
Oh I see, you collapse them into `||` but I think that makes it harder to read.  I'd rather keep the two blocks for readability.